ckan/lib/activity_streams_session_extension.py

```diff
@@ -70,8 +70,8 @@ def before_commit(self, session):
             # object is a package.
 
             # Don't create activities for private datasets.
-            if obj.private:
-                continue
+            # if obj.private:
```

Comment:
That's quite a big change, was it intended? I've always thought that private datasets should have activity streams too, only hidden to non authorized users. 
I guess the original issue was that if the dataset was eventually made public the previously private activities would be made public too. I'm not sure I find that an issue but if it is we could add a functionality to give an option to editors to clear the previous activities when making a dataset public.

@ckan/core what are your views on activities and private datasets?

Reply:
Agree the simple solution is just to ask at the time the dataset has its visibility changed, but the default action should be to just not touch the visibility of activity events rather than delete them. Activity is important for auditing for many groups.

Quick solution, add a migration along with this that adds a `private` boolean to each activity event and populates it with the value from the dataset at the time of migration. Activity events use the same logic as the dataset to show or hide if `private` is set.

Editors would be able to show or hide activity events individually or in bulk, but culling would requirer sysadmin?

Reply:
The PR definitely doesn't need private datasets to have activity streams - my organization mostly uses private datasets, so I made the change with these in mind, but it isn't necessary. I think the migration solution sounds good but I will probably not be able to implement it soon so it would be fine to not save activity data for private datasets until then.

Reply:
Cool, in which case I suggest this is change is removed from this PR and it can be tackled in a separate PR